cfme/automate/dialogs/__init__.py

```diff
@@ -54,8 +55,8 @@ class EditDialogView(DialogForm):
     @property
     def is_displayed(self):
         return (
-            self.in_customization and self.service_dialogs.is_opened and
-            self.label.text == self.label
+            self.in_customization and self.title.text == "Automate Customization" and
+            self.sub_title.text == "General" and bool(self.label.read())
```

Comment:
What's the purpose of the `bool(self.label.read())` check? just that the label field is populated?

I think you need to check that `self.label.read() == self.context['object'].label` - that the content of the label field matches the Dialog's `label` attribute.  If this can be changed during edit, then this may be a weak check that we're on the correct page. It may be better to just `return False` for this method, or raise NotImplemented per the change coming in #7958

Reply:
Looking at the UI a bit, I don't think we should try to match this edit view via is_displayed, this will not be robust. The label can be modified, navigation to this edit page should always be performed via its prerequisite tree.

Reply:
@mshriver rather than removing all `is_displayed` checks better to go with check for label (`TextInput` box).
1. If we go for add page `TextInput` box read will be blank.
2. If we go for edit then `TextInput` box read will be `old name`.

I think those check is sufficient to check for edit view.

Reply:
@niyazRedhat it looks like you ended up changing this to matching the object label?

I think there's an assumption in the check you were suggesting above - that the page has just been navigated to.

It's possible the browser is on an Add page, that has been partially filled in. If we tried to navigate to Edit, and were only checking that the label field is filled, it would short-circuit navigation and think it was on the right page.